server/conf/routes

```diff
@@ -25,6 +25,8 @@ POST    /admin/programs/:programId/admins       controllers.admin.ProgramAdminMa
 
 # Routes for a CiviForm admin to manage application statuses for a given program.
 GET     /admin/programs/:programId/statuses controllers.admin.AdminProgramStatusesController.index(request: Request, programId: Long)
+POST    /admin/programs/:programId/statuses controllers.admin.AdminProgramStatusesController.index(request: Request, programId: Long)
```

Comment:
I don't think "index" actions should be POSTs, as well we shouldn't double up GET/POST on the same handler.  I don't think that's RESTful or something I've seen us do otherwise; though I could be wrong.

Reply:
Since application statuses don't have a "true" identifier that we want to put in a URL, I don't think we'd want to create anything REST-ful for statuses.

GET/POST on the same handler is definitely something I've seen done before since you're updating the entity being referred to (in this case the program's statuses). This approach falls out from the fact that:
  1) We want to have a single page with modal forms for editing individual stauses (from agreed-on UX)
  2) When form validation errors occur, it'd be unfortunate to keep the user on a different URL than they started and still render the same content (e.g. URL is `/update` but still renders full index page
  3) REST-ful APIs in fact would use the same collection identifier with different HTTP verbs to indicate whether it corresponded to a CREATE or UPDATE (https://restfulapi.net/http-methods/). In this case, we merge CREATE / UPDATE into a single `POST` rather than having an overload that accepts a request with verb `PATCH`.

Let me know what you think. I'm also happy to create two separate `AdminProgramStatusesController.[index|update]` methods if you think that'd aid in readability. Under the hood, they'll still call largely the same underlying logic in the controller and the same view.

Reply:
Note that we could use another URL to support the POST, but we'd eventually want to redirect back to the GET version of the URL. Performing this redirect would mean we'd lose the POST data and we'd need to preserve it by putting it in the `.flash` ([example](https://github.com/seattle-uat/civiform/blob/cfbc4a9eff6099cf7adf6c2283f55c037eb033dd/server/app/controllers/ti/TrustedIntermediaryController.java#L136-L143)).

This approach felt more error-prone to me if we end up updating what POST data is required / etc. My belief is that very very few people outside of devs on a project are actually looking at the specific URL as they're submitting forms, so I want to make sure there's a clear win for the user here.